python-lib/dku_stress_test_center/metrics.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+from sklearn.metrics import accuracy_score
+
+def get_performance_metric(metric_name: str):
+    return accuracy_score # TODO
+
+def greater_perf_is_better(metric_name: str):
+    return True # TODO
+
+def worst_group_performance(metric_name: str, y_true: np.array, y_pred: np.array):
```

Comment:
This is correct, but I would make it more general. In the sense that we also give in input the identifiers of the subpopulation each sample belongs to. Could be  `worst_group_performance(metric_name: str, y_true: np.array, y_pred: np.array, subpop_id: np.array)`, so that we can use it also when subpopulations are not defined by the classes. In this specific case, we would have in input a `subpop_id=y_true` (which could be the default, btw), but this is a special case. WDYT?

Reply:
Yep sounds nice indeed!

Reply:
Actually side note, the subpop_id array would still have to be either a subset/superset of the target classes, since we use the array to define the mask `class_mask = y_true == target_class`. Like it must have at least some unique values in common with `y_true` otherwise we do not compute anything

Reply:
So `subpop_id` would contain kind of cluster_ids, independent from the target classes. It represents any partition of the data samples. I think the implementation should be changed to something like:

```
subpop_groups = np.unique(subpop_id)
for subpop_group in subpop_groups:
    group_mask = subpop_id == subpop_group
    performance = get_performance_metric(metric_name)(y_true[group_mask], y_pred[group_mask])
    performances.append(performance)
```

Reply:
when` subpop_id=y_true` we fallback to your implementation is I'm not mistaken.

Reply:
Done in e507cfe